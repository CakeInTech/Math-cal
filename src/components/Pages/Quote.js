import React, { useState } from 'react';
import './page.css';

function Quote() {
  const [state, setState] = useState({
    id: 1,
    author: 'Public Quotes',
    quote: 'Mathematics is the queen of science, and arithmetic the queen of mathematics.',
  });

  // Generate randmon Quotes

  const quoteApi = async () => {
    const response = await fetch('https://random-math-quote-api.herokuapp.com/');

    const { id, author, quote } = await response.json();

    setState({ id, author, quote });
    return { id, author, quote };
  };
  const { author, quote } = state;
  return (
    <div className="quoteContainer">
      <div className="generator">
        <figure>
          <blockquote>
            <p>
              &quot;
              {quote}
              {' '}
              &quot;
            </p>
          </blockquote>
          <figcaption>
            â€”
            {author}
          </figcaption>
        </figure>

      </div>
      <button
        className="btn btn-outline-info"
        type="button"
        onClick={quoteApi}
      >
        Get Random Quote
      </button>
    </div>
  );
}

export default Quote;
