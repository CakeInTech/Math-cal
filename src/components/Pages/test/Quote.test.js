import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../Quote';

it('renders correctly', () => {
  const tree = renderer.create(<Quote />);
  expect(tree).toMatchSnapshot();
});
