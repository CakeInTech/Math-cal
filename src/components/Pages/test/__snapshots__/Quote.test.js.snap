// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="quoteContainer"
>
  <div
    className="generator"
  >
    <figure>
      <blockquote>
        <p>
          "
          Mathematics is the queen of science, and arithmetic the queen of mathematics.
           
          "
        </p>
      </blockquote>
      <figcaption>
        â€”
        Public Quotes
      </figcaption>
    </figure>
  </div>
  <button
    className="btn btn-outline-info"
    onClick={[Function]}
    type="button"
  >
    Get Random Quote
  </button>
</div>
`;
